transaction/std/src/transaction_builder.rs

```diff
@@ -3104,4 +3105,505 @@ pub mod transaction_builder_tests {
+                    subaddress_matches_tx_out(&sender, GIFT_CODE_SUBADDRESS_INDEX, tx_out).unwrap()
+                })
+                .expect("Didn't find gift code funding output");
+
+            let funding_change_output = funding_tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|tx_out| {
+                    subaddress_matches_tx_out(&sender, CHANGE_SUBADDRESS_INDEX, tx_out).unwrap()
+                })
+                .expect("Didn't gift code funding change output");
+
+            validate_tx_out(block_version, funding_output).unwrap();
+            validate_tx_out(block_version, funding_change_output).unwrap();
+
+            assert!(
+                subaddress_matches_tx_out(&sender, GIFT_CODE_SUBADDRESS_INDEX, funding_output)
+                    .unwrap()
+            );
+            assert!(subaddress_matches_tx_out(
+                &sender,
+                CHANGE_SUBADDRESS_INDEX,
+                funding_change_output
+            )
+            .unwrap());
+
+            // Ensure funding output & change memos are correct
+            let funding_output_public_key =
+                &RistrettoPublic::try_from(&funding_output.public_key).unwrap();
+            let funding_change_output_tx_out_public_key =
+                &RistrettoPublic::try_from(&funding_change_output.public_key).unwrap();
+            let funding_output_ss =
+                get_tx_out_shared_secret(sender.view_private_key(), funding_output_public_key);
+            let funding_change_output_ss = get_tx_out_shared_secret(
+                sender.view_private_key(),
+                funding_change_output_tx_out_public_key,
+            );
+
+            let (funding_amount, _) = funding_output
+                .masked_amount
+                .get_value(&funding_output_ss)
+                .unwrap();
+            assert_eq!(funding_amount.value, funding_output_amount.value);
+            assert_eq!(funding_amount.token_id, token_id);
+
+            if block_version.e_memo_feature_is_supported() {
+                let funding_change_output_memo = funding_change_output
+                    .e_memo
+                    .unwrap()
+                    .decrypt(&funding_change_output_ss);
+                let funding_output_memo =
+                    funding_output.e_memo.unwrap().decrypt(&funding_output_ss);
+                match MemoType::try_from(&funding_change_output_memo)
+                    .expect("Couldn't decrypt funding change memo")
+                {
+                    MemoType::GiftCodeFunding(memo) => {
+                        assert!(memo.public_key_matches(funding_output_public_key),);
+                        assert_eq!(memo.funding_note().unwrap(), note,);
+                    }
+                    _ => {
+                        panic!("unexpected memo type")
+                    }
+                }
+                match MemoType::try_from(&funding_output_memo)
```

Comment:
This can be shortened to `assert!(matches!(` (or `assert_matches!` is thats already a dependency of this crate)